### DISCUSSION TITLE
OctoPrint Bed Level Visualizer -- what gcode for Sidewinder?

### DISCUSSION THREAD

**Discussion by Ltek** [jneilliii/OctoPrint-BedLevelVisualizer#541] [Q&A]
For the Sidewinder gcode... are we supposed to use both the G20 (bilinear or unified) AND add/change the according to the info in the Sidewinder section on that same page?

thx for the help, the instructions are not clear.

**Comment by jneilliii** [comment]
it's going to be whatever your firmware uses. I don't have a Sidewinder or know what it was flashed with from factory or what you configured if you personally flashed it. The most common ABL method used in manufacturer's firmware is bilinear from what I've seen in general.

**Reply by Ltek** [reply]
I was referring to this page... https://github.com/jneilliii/OctoPrint-BedLevelVisualizer/blob/master/wiki/gcode-examples.md 
Do I use Bilinear then add/update that using the Sidewinder gcode lower on that page?

**Comment by jneilliii** [comment]
Oh, I apologize. No, if you have a sidewinder you just use the sidewinder example. It was provided by a user that had that printer. If you look close, it's technically the same as the bilinear example at it's core. You can get away with something as simple as this

```
G28
@BEDLEVELVISUALIZER
G29
```
but the additional bits before and after on the examples do additional things like heat up bed/nozzle before and after turn those off as well as steppers.

**Reply by Ltek** [reply]
gotcha, thank you!
